### PR TITLE
Clean up GitHub connector

### PR DESCRIPTION
**Pull request by diminishedprime**

### CODE REVIEW

**Comment by diminishedprime** [comment]
> I'm afraid I don't understand how the updated connector distinguishes between stars mode and issues mode; can you elaborate a bit?

We determine which endpoint to call by checking the `group` key in the schema. The stars fields are all under the `stargazers' group and the issue fields are all under the `issues` group. If  you pick fields that are from two separate groups, and error is thrown explaining that.

File: github/src/main.js [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+  var response = {
+    schema: requestedFields.build(),
+    rows: rows,
+  };
+  return response;
+}
+
+function parseIssueRow(requestedFields, issue) {
+  var row = [];
+  requestedFields.asArray().forEach(function(requestedField) {
```
[the remark this answers is not in the document]
Comment (diminishedprime) [reply github/src/main.js]: Yep. Done.

File: github/src/main.js [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+      return cc
+        .newUserError()
+        .setText('Fields from group: "' + endpoint + '" are not supported.')
+        .throwException();
+  }
+}
+
+function parseStarRow(requestedFields, star) {
+  var row = [];
+  requestedFields.asArray().forEach(function(requestedField) {
```
Comment (Lucretiel) [inline github/src/main.js]: This should also be a map

**Comment by diminishedprime** [comment]
> Do we care about docstrings on these functions? Is this code intended to be a reference for community best practice?

No, this one won't be a reference for community best practice so we're not worried about docstrings.

**Pull request merged**